Fix 18109: Specifying atmost chars in topic name, description, web fra…

Shivanshu-09 opened the pull request:
…gment

## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #18109 .
2. This PR modified the **topic-editor-tab-directive.ts** file to define maximum chars and min char variables.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->

#### Proof of changes on desktop.
<img width="1680" alt="Screenshot 2023-04-23 at 9 14 15 PM" src="https://user-images.githubusercontent.com/61676272/233853405-c94194af-ea9b-4666-a576-828f7415aad1.png">


<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

shivanshu-sarin committed:
Fix-18109 Specifying atmost chars in topic name, description, web fragment

shivanshu-sarin committed:
Fix-18109 Specifying atmost chars in topic name, description, web fragment

Shivanshu-09 commented:
Hi @Rijuta-s, I have corrected the checks. Can you please review.

Rijuta-s commented on core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.ts:
```diff
@@ -92,6 +92,11 @@ export class TopicEditorTabComponent implements OnInit, OnDestroy {
   uncategorizedEditOptionsIndex: number;
   subtopicEditOptionsAreShown: number;
   skillOptionDialogueBox: boolean = true;
+  MAX_CHARS_IN_TOPIC_NAME: number;
```
Where is this being used?

Shivanshu-09 replied:
All these variables are defined in **constraints.ts.**

Rijuta-s replied:
Can you please provide me with the path to constraints.ts file, or add a snap shot of it? Also can you add a comment above the list of variables that are defined in constraint.ts that these variables are defined in constraints.ts file?

Shivanshu-09 replied:
Path to constraints.ts file = asserts/constraints.ts. Adding SS of place where these variables have been defined.
<img width="1680" alt="Screenshot 2023-04-29 at 1 17 39 AM" src="https://user-images.githubusercontent.com/61676272/235239796-31451ecc-50ea-462b-8098-ed60f7259676.png">

Rijuta-s commented on core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.ts:
```diff
@@ -92,6 +92,11 @@ export class TopicEditorTabComponent implements OnInit, OnDestroy {
   subtopicEditOptionsAreShown: number;
   skillOptionDialogueBox: boolean = true;
+  MAX_CHARS_IN_TOPIC_NAME: number;
+  MAX_CHARS_IN_TOPIC_DESCRIPTION: number;
```
Ditto as above

Rijuta-s commented on core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.ts:
```diff
@@ -92,6 +92,11 @@ export class TopicEditorTabComponent implements OnInit, OnDestroy {
+  MAX_CHARS_IN_TOPIC_NAME: number;
+  MAX_CHARS_IN_TOPIC_DESCRIPTION: number;
+  maxWebTitleFrag: number;
+  minWebTitleFrag: number;
```
Where are these being assigned values?

Shivanshu-09 replied:
These were previously directly assigned in **topic-editor-tab.directive.html** . But these values must be defined in JS file also, so I added them in **topic-editor-tab.directive.ts** file.

Rijuta-s commented on core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.ts:
```diff
@@ -92,6 +92,11 @@ export class TopicEditorTabComponent implements OnInit, OnDestroy {
+  MAX_CHARS_IN_TOPIC_DESCRIPTION: number;
+  maxWebTitleFrag: number;
+  minWebTitleFrag: number;
+  MAX_CHARS_IN_META_TAG_CONTENT: number;
```
Where is this being used?

Shivanshu-09 replied:
These are used in **topic-editor-tab.directive.html**

Shivanshu-09 replied:
Hi @Rijuta-s Is there anything else needed for the PR to be merged?

Rijuta-s reviewed (changes requested):
Left a few comments, please take a look.

Shivanshu-09 commented:
I have updated the PR with latest develop changes. @Rijuta-s can you direct me if I am doing something wrong. This is my first contribution to this project, can you provide starting help please. Thank you.

Shivanshu-09 committed:
Merge branch 'develop' into Fix-18109

StephenYu2018 committed:
Fix issue #17983: disable the SAVE button after its first click when creating a new question (#18131)

Change questionIsBeingSaved instance variable purpose to disable the SAVE button only when a question is being saved

won369369 committed:
Fix part of #14219: Increase backend test coverage of learner_goals_services_test to 100% (#17795)

* added test_remove_topics_when_learner_goals_model_does_not_exist

* Solved line too long lint error

* wrote test_remove_topics_from_learn_goal_executed_correctly

* Created test_get_learner_goals_from_model_called

* mock import lint error

* Added comment for using object

* trailing whitespace

* Corrected function name

* tested if get_learner_goals_from_model has not been executed

* Changed mock.patch usage to try/exception

* Solved lint errors

* simply call the function and then check the state of the learner_goals

gp201 commented:
@Rijuta-s PTAL

Rijuta-s commented:
@Shivanshu-09 some tests are failing, please fix them and also, please assign me once the changes from your end is done, otherwise I might miss reviewing the PR. Thanks

Uglycrown committed:
Fix #18075: Chromevox is not reading label for button '+' when focus is on it in Topics dashboard (#18146)

* Now Chromevox can read close button for x icon after uploading a picture

* label-fix-chromvox

* fix-ffa

* fix-chromevox-read-create-new-topic

shivanshu-sarin committed:
Merge branch 'Fix-18109' of github.com:Shivanshu-09/oppia into Fix-18109

Shivanshu-09 commented:
I don't have permission to assign I guess. But I have replied to your comments and checks also are passing @Rijuta-s .
Thanks!

seanlip commented on core/domain/learner_goals_services_test.py:
```diff
@@ -256,3 +256,62 @@ def test_get_all_topic_ids_in_learn(self) -> None:
             learner_goals_services.get_all_topic_ids_to_learn(
                 self.viewer_id), [self.TOPIC_ID_1, self.TOPIC_ID_2])
+
+    def test_remove_topics_from_learn_goal_executed_correctly(
```
Why is this in the PR? It has nothing to do with this PR as far as I can tell.

seanlip commented on core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.html:
```diff
@@ -111,7 +111,7 @@ <h3 class="topic-card-header">Details</h3>
                         class="form-control topic-description e2e-test-topic-page-title-fragment-field"
                         [(ngModel)]="editablePageTitleFragmentForWeb"
-                        [maxlength]="MAX_CHARS_IN_PAGE_TITLE_FRAGMENT_FOR_WEB"
+                        [maxlength]="maxWebTitleFrag"
```
Please don't abbreviate words, and choose your variable names more carefully -- I don't know what a "maxWebTitleFrag" is.

With max/min there needs to be some idea of what the max/min is being applied to (in this case, characters) but that's not clear from the variable name. Please revise, here and elsewhere.

seanlip reviewed (changes requested):
@Shivanshu-09 Your PR contains a lot of extraneous changes (see the "Files Changed" tab). Could you please fix it so that it contains only the correct changes that are needed for this PR? (You can check the changes yourself by going to the "Files Changed" tab -- please make sure this is correct before reassigning reviewers.)

Thanks!
